src/multivariate/mvnormalcanon.jl

```diff
@@ -118,3 +118,5 @@ unwhiten_winv!(J::ScalMat, x::AbstractVecOrMat) = whiten!(J, x)
 
 _rand!(rng::AbstractRNG, d::MvNormalCanon, x::AbstractMatrix) = add!(unwhiten_winv!(d.J, randn!(rng,x)), d.μ)
 _rand!(d::MvNormalCanon, x::AbstractMatrix) = _rand!(Base.GLOBAL_RNG, d, x)
+_rand!(rng::AbstractRNG, d::MvNormalCanon, x::AbstractVector) = add!(unwhiten_winv!(d.J, randn!(rng,x)), d.μ)
```

Comment:
Since these are identical to the matrix methods, I'd just change the signature of the previous methods to accept `AbstractVecOrMat`.